### PR TITLE
Prevent generation of default accounts

### PR DESCRIPTION
**Pull request by crufter**
Notable changes
- Only generate default account in default namespace
- Added `DisableAdmin` flag to even disable this behaviour. This is useful for security hardening purposes, ie. imagine data loss occuring and then an attacker being able to log in with default credentials.
- Added `--namespace` option to `micro auth account create` as test which previously depended on account being autogenerated after namespace change now must create an admin account in that namespace first.

- [x] Fix
- [x] Test

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -126,6 +126,11 @@ var (
 			EnvVars: []string{"MICRO_AUTH_PRIVATE_KEY"},
 			Usage:   "Private key for JWT auth (base64 encoded PEM)",
 		},
+		&cli.BoolFlag{
+			Name:    "no_default_account",
```
Comment (asim) [inline cmd/cmd.go:130]: --disable_admin / MICRO_AUTH_DISABLE_ADMIN

File: cmd/cmd.go [hunk 2]
```diff
@@ -127,6 +127,11 @@ var (
 			EnvVars: []string{"MICRO_AUTH_PRIVATE_KEY"},
 			Usage:   "Private key for JWT auth (base64 encoded PEM)",
 		},
+		&cli.BoolFlag{
```
Comment (ben-toogood) [inline cmd/cmd.go:130]: Please add this as a flag in service/auth

File: service/auth/server/auth/auth.go [hunk 3]
```diff
@@ -59,6 +61,12 @@ func (a *Auth) Init(opts ...auth.Option) {
 }
 
 func (a *Auth) setupDefaultAccount(ns string) error {
+	if ns != namespace.DefaultNamespace {
```
Comment (ben-toogood) [inline service/auth/server/auth/auth.go:64]: Why is this check required? Doesn't this circumvent the need for DisableAdmin flag?

**Review by ben-toogood — CHANGES_REQUESTED** [review]

File: service/auth/server/auth/auth.go (hunk 3 above)
Comment (crufter) [reply service/auth/server/auth/auth.go:64]: No. Admins from now on only get generated for the default account.

File: cmd/cmd.go (hunk 2 above)
Comment (crufter) [reply cmd/cmd.go:130]: Why? I don't see the `MICRO_AUTH_PRIVATE_KEY` flags there and it's a configuration option just like `MICRO_AUTH_PRIVATE_KEY`?

Comment (ben-toogood) [reply cmd/cmd.go:130]: It is specific to the auth server, MICRO_AUTH_PRIVATE_KEY and MICRO_AUTH_PRIVATE_KEY are used by all core components.

File: service/auth/server/auth/auth.go (hunk 3 above)
Comment (ben-toogood) [reply service/auth/server/auth/auth.go:64]: So if DisableAdmin is set, how do we ever create the first root account?

Comment (crufter) [reply service/auth/server/auth/auth.go:64]: We don't. See ticket description

> Added DisableAdmin flag to even disable this behaviour. This is useful for security hardening purposes, ie. imagine data loss occuring and then an attacker being able to log in with default credentials.

File: cmd/cmd.go (hunk 2 above)
Comment (crufter) [reply cmd/cmd.go:130]: Makes sense, thanks

**Review by ben-toogood — APPROVED** [review]

**Pull request merged**